src/compiler/core.ts

```diff
@@ -778,19 +778,23 @@ namespace ts {
         return [] as any as SortedArray<T>; // TODO: GH#19873
     }
 
-    export function insertSorted<T>(array: SortedArray<T>, insert: T, compare: Comparer<T>, allowDuplicates?: boolean): void {
+    export function insertSorted<T, U>(array: SortedArray<T>, insert: T, keySelector: (v: T) => U, keyComparer: Comparer<U>, allowDuplicates?: boolean): boolean {

```

Comment:
What does this return now? Whether a duplicate element was returned?

Reply:
That's correct, yes. I'll be sure to document this if/when I rebase and clean up this PR.